app/drivers/tools/repair/java/ET.py

```diff
@@ -0,0 +1,147 @@
+
+    def run_repair(self, bug_info, repair_config_info):
+        super(ET, self).run_repair(bug_info, repair_config_info)
+        self.timestamp_log_start()
+
+        print('!!! begin')
+        #return #####
+
+        assert bug_info['language']=='java'
+        assert len(bug_info['failing_test'])>0
+
+        """
+            self.dir_logs - directory to store logs
+            self.dir_setup - directory to access setup scripts
+            self.dir_expr - directory for experiment
+            self.dir_output - directory to store artifacts/output
+        """
+
+        repo_path = (Path(self.dir_expr) / 'src').resolve()
+        setup_path = Path(self.dir_setup).resolve()
+        #print(bug_info, repair_config_info, self.container_id)
+
+        # test list maybe `com.clz::mtd` or `com.clz`, let's make them into `com.clz`
+
+        test_failed = []
+        test_failed_set = set()
+        for t in bug_info['failing_test']:
+            t = t.partition('::')[0]
+            if t not in test_failed_set:
+                test_failed_set.add(t)
+                test_failed.append(t)
+
+        test_passed = []
+        test_passed_set = set()
+        for t in bug_info['passing_test']:
+            t = t.partition('::')[0]
+            if t not in test_failed_set and t not in test_passed_set:
+                test_passed_set.add(t)
+                test_passed.append(t)
+
+        self.write_json({
+            'id': int(bug_info['id']),
+            'repo_path': str(repo_path),
+            'setup_script_path': str(setup_path),
+
+            'sp_src': bug_info['source_directory'],
+            'sp_test': bug_info['test_directory'],
+            'tp_src': bug_info['class_directory'],
+            'tp_test': bug_info['test_class_directory'],
+            'cp_compile': ':'.join([str(Path(self.dir_expr)/s) for s in bug_info['dependencies']]),
+            'cp_test': ':'.join([
+                str(repo_path/bug_info['class_directory']),
+                str(repo_path/bug_info['test_class_directory']),
+                *[str(Path(self.dir_expr)/s) for s in bug_info['dependencies']],
+            ]),
+            'lang_level': bug_info['java_version'],
+
+            'test_passed': test_passed,
+            'test_failed': test_failed,
+            'test_timeout': bug_info['test_timeout'],
+            'test_sh_fn': bug_info['test_script'],
+
+            'total_timeout_s': int(float(repair_config_info['timeout'])*3600),
+            'cpu_count': len(repair_config_info['cpus']),
+            'gpu_count': len(repair_config_info['gpus']),
+        }, '/root/workflow/info.json')
+
+        ret = self.run_command('bash -c "python3 /root/workflow/main.py"', log_file_path='/root/workflow/log.txt')
+
+        #print(*self.read_file('/root/workflow/log.txt'), sep='')
+
+        self.process_status(ret)
+        self.timestamp_log_end()
+
+        print('!!! end')
```

Comment:
Hi, small thing. You should preferably use - self.emit_... so that this works in GUI mode and commandline mode

Reply:
Okay this is only used in my local debugging. I also tweaked several other files so that it can instantly output command's STDOUT (instead of waiting for it to finish). I will comment out this if it breaks the GUI mode.

Reply:
Oh nono, it does not break the GUI mode :smile: that I can say from experience